File: src/main/java/ru/anarcom/octopus/entity/basic/BaseEntity.java
package ru.anarcom.octopus.entity.basic;

import java.util.Date;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

@MappedSuperclass
@Data
@EqualsAndHashCode
public class BaseEntity {

  @Id
  @GeneratedValue(strategy= GenerationType.AUTO)
  protected Long id;

  @Enumerated(EnumType.STRING)
  EntityStatus status;

  @CreatedDate
  private Date createdAt;

  @LastModifiedDate
  private Date updatedAt;
}

File: src/main/kotlin/ru/anarcom/octopus/entity/basic/BaseEntity.kt
package ru.anarcom.octopus.entity.basic

import lombok.experimental.Accessors
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import java.util.*
import javax.persistence.*

@MappedSuperclass
@Accessors(chain = true)
open class BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null

    @Enumerated(EnumType.STRING)
    var status: EntityStatus = EntityStatus.NOT_ACTIVE

    @CreatedDate
    val createdAt: Date? = null

    @LastModifiedDate
    val updatedAt: Date? = null
}